File: aula3/ping.py
# Bibliotecas
from pymongo.mongo_client import MongoClient
from pymongo.server_api import ServerApi
from urllib.parse import quote_plus

# "Escapar" o nome e a senha
usuario = quote_plus("seuusuario")
senha = quote_plus("suasenha")

# URL
url = f"mongodb+srv://{usuario}:{senha}@cluster1.tjqkv.mongodb.net/?retryWrites=true&w=majority&appName=cluster1"

# Criar um novo cliente e conectar ao servidor
client = MongoClient(url, server_api = ServerApi('1'))

# Enviar a solicitação e confirmar a conexão
try:
    client.admin.command('ping')
    print("Sucesso ao conectar!")
except Exception as e:
    print(e)

File: aula11/ping.js
// Bibliotecas
const express = require('express');
const mongoose = require('mongoose');
const app = express();

// Escape o nome do usuário e a senha
const usuario = encodeURIComponent('paulosermoreira');
const senha = encodeURIComponent('teste@92');

// URI de conexão
const uri = `mongodb+srv://${usuario}:${senha}@cluster1.tjqkv.mongodb.net/?retryWrites=true&w=majority&appName=cluster1`;

//Função para conectar ao MongoDB
async function conectarMongoDB() {
    try {
        await mongoose.connect(uri);
        console.log('Sucesso ao conectar ao MongoDB');
        return true;
    } catch(erro) {
        console.error('Erro ao conectar ao MongoDB:', erro);
        return false;
    }
}

// Rota para testar a conexão
app.get('/ping', async(req, res) => {
    const conectado = await conectarMongoDB();
    if(conectado) {
        res.send('Sucesso ao conectar ao MongoDB');
    } else {
        res.status(500).send('Erro ao conectar ao MongoDB');
    }

    // Fechar a conexão
    mongoose.connection.close();
});

// Iniciar a aplicação
const porta = 3000;
app.listen(porta, () => {
    console.log(`Servidor rodando em http://localhost:${porta}`);
});